.github/workflows/event-processor.yml

```diff
@@ -0,0 +1,155 @@
+name: GitHub Event Processor
+
+on:
+  issues:
+    types: [edited, labeled, opened, reopened, unlabeled]
+  # issue_comment is used for both issues and pull_requests
+  # github.event.issue.pull_request will be non-null on pull request comments
+  issue_comment:
+    types: [created]
+  # synchronize is the pull_request_target event when changes are pushed
+  # pull request merged is the closed event with github.event.pull_request.merged = true
+  pull_request_target:
+    types: [closed, labeled, opened, reopened, review_requested, synchronize, unlabeled]
+
+# This removes all unnecessary permissions, the ones needed will be set below.
+# https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+permissions: {}
+
+jobs:
+  # This event requires the Azure CLI to get the LABEL_SERVICE_API_KEY from the vault.
+  # Because the azure/login step adds time costly pre/post Az CLI commands to any every job
+  # it's used in, split this into its own job so only the event that needs the Az CLI pays
+  # the cost.
+  event-handler-with-azure:
+    permissions:
+      issues: write
+      pull-requests: write
+      # For OIDC auth
+      id-token: write
+      contents: read
+    name: Handle ${{ github.event_name }} ${{ github.event.action }} event with azure login
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+    steps:
+      - name: 'Az CLI login'
+        uses: azure/login@v2
+        with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
```

Comment:
We will need to setup these secrets before this action will work.

Reply:
We will need to wait until @benbp is back as he didn't grant me owner to the app registration we are using here. https://ms.portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/~/Owners/appId/a6dd2dfe-7352-41a7-9020-05301c3bca1a/isMSAApp~/false

Reply:
Added @weshaggard and @mikeharder as owner on this.